eth2/utils/merkle_proof/src/lib.rs

```diff
@@ -328,4 +402,45 @@ mod tests {
         assert!(verify_merkle_proof(leaf, &[], 0, 0, leaf));
         assert!(!verify_merkle_proof(leaf, &[], 0, 7, junk));
     }
+
+    #[test]
+    fn push_complete_example() {
```

Comment:
I couldn't resist writing a quickcheck test for this as well: https://github.com/sigp/lighthouse/commit/492614018b6c7dd17cf24631b3433a97d040dc22

Reply:
Oh thanks! I had seen quickcheck being used in another crate, but wasn't sure I understood it correctly... ! Thanks :)